model/vcs/metrics.yaml

```diff
@@ -0,0 +1,147 @@
+  - id: metric.vcs.change.time_to_approval
+    type: metric
+    metric_name: vcs.change.time_to_approval
+    brief: 'The amount of time it took a change (pull request/merge request/changelist) to go from open to approved'
+    instrument: gauge
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.name
+        requirement_level: required
+  - id: metric.vcs.repository.count
+    type: metric
+    metric_name: vcs.repository.count
+    brief: 'The number of repositories in an organization'
+    instrument: updowncounter
+    unit: "{repository}"
+    stability: experimental
+    attributes: []
+  - id: metric.vcs.ref.count
+    type: metric
+    metric_name: vcs.ref.count
+    brief: 'The number of refs of type branch or tag in a repository'
+    instrument: updowncounter
+    unit: "{ref}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.type
+        requirement_level: required
+  - id: metric.vcs.change.lines_delta
+    type: metric
+    metric_name: vcs.change.lines_delta
+    brief: 'The number of lines added/removed in a change relative to the target branch of the change'
+    instrument: gauge
+    unit: "{line}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.name
+        requirement_level: required
+      - ref: vcs.ref.type
+        requirement_level: required
+      - ref: vcs.line_change.type
+        requirement_level: required
+  - id: metric.vcs.ref.lines_delta
+    type: metric
+    metric_name: vcs.ref.lines_delta
+    brief: 'The number of lines added/removed in a ref (branch) relative to the previous day'
+    instrument: gauge
+    unit: "{line}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.name
+        requirement_level: required
+      - ref: vcs.ref.type
+        requirement_level: required
+      - ref: vcs.line_change.type
+        requirement_level: required
+  - id: metric.vcs.change.revisions_delta
+    type: metric
+    metric_name: vcs.change.revisions_delta
+    brief: 'The number of revisions (commits) a change is ahead/behind the target branch of the change'
+    instrument: gauge
+    unit: "{revision}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.name
+        requirement_level: required
+      - ref: vcs.ref.type
+        requirement_level: required
+      - ref: vcs.revision_delta.type
+        requirement_level: required
+  - id: metric.vcs.ref.revisions_delta
+    type: metric
+    metric_name: vcs.ref.revisions_delta
+    brief: 'The number of revisions (commits) a ref (branch) is ahead/behind the default branch (trunk)'
```

Comment:
let's say we're two commits ahead and 3 behind

![image](https://github.com/user-attachments/assets/f7fc079a-7f6c-4ba5-9a7c-fc801160c09f)

would we record 2 measurements for this metric with different `vcs.revision_delta.type`?

Could you please a note and elaborate on how someone should capture this?

Reply:
I don't think we need both `change.revision_delta` and `ref.revision_delta`. I think we only need `ref.*` because a change is always a ref. Basically the same comment as [this comment](https://github.com/open-telemetry/semantic-conventions/pull/1383/files?short_path=9b51d2f#r1824428894).

To your question @lmolkova - I think this [other comment](https://github.com/open-telemetry/semantic-conventions/pull/1383/files?short_path=9b51d2f#r1824440816) also talks about that. This too was originally two separate metric names instead of one metric with variations of an attribute.

But you will always have to look at the state of the branch. This is how I do it in code from the GitHub receiver currently. https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/githubreceiver/internal/scraper/githubscraper/github_scraper.go#L140-L141

Reply:
Thanks! That's what I wanted to clarify. So maybe we can explain it in semconv?

Something like 

```suggestion
    brief: 'The number of revisions (commits) a ref (branch) is ahead/behind the default branch (trunk)'
    note: This metric should be reported for each `vcs.ref.revisions_delta` value. For example if branch `a` is 3 commits behind and 2 commits ahead of `trunk`, instrumentation SHOULD report two measurements: 3 and 2.
```

Reply:
I such a added note for the line and revision metrics in 5d5a56b2776f2da0323add3f92c5740917f0204c.
I also specified that the measurements should be positive.

> I don't think we need both change.revision_delta and ref.revision_delta. I think we only need ref.* because a change is always a ref.

We decided in the CICD SemConv SIG meeting of the October 3rd to have separate metrics because the semantics are slightly different between ref and change. For vcs.ref.revision_delta we always compare against the default branch, while for vcs.change.revision_delta we compare against the target branch. The target branch of a change is usually the default branch, but not necessarily so.

Reply:
I proposed in https://github.com/open-telemetry/semantic-conventions/pull/1383#pullrequestreview-2411906073 what combining the change and ref metrics for `revisions_delta` and `lines_delta` could look like.
If that looks good to you I will make those changes in the code.

Reply:
See this [comment](https://github.com/open-telemetry/semantic-conventions/pull/1383/files#r1827082069) as well.

Reply:
I have combined `vcs.ref.lines_delta` and `vcs.change.lines_delta` in 8f8bb73943ce328f172f402b0683b45412073c2f.